### ISSUE TITLE
Add email verification and change password

### ISSUE THREAD

**Issue by shnupta** [fuse-compound/Fuse.Firebase#83]
Would it be possible to implement the functions to send a verification email and also have the option to change the user's password?

**Comment by LuisRodriguezLD** [comment]
Got them, they work great on iOS and they _should_ work on Android. Unfortunately I do not have one, give them a go if they are good to go I'll merge them to master. Try out [this branch].(https://github.com/fuse-compound/Fuse.Firebase/tree/new-auth-features)

**Comment by shnupta** [comment]
I'll give it a test later tonight and come back to you!

**Comment by shnupta** [comment]
Getting a build error at the moment:

```
/Users/caseywilliams/Fuse/QuickLaps/build/Android/Debug/app/src/main/java/com/foreign/Firebase/Authentication/Email/SendVerificationEmail.java:28: error: cannot find symbol
        final FirebaseUser user = (FirebaseUser)com.foreign.Firebase.Authentication.User.GetCurrent412();
              ^
  symbol:   class FirebaseUser
  location: class SendVerificationEmail
/Users/caseywilliams/Fuse/QuickLaps/build/Android/Debug/app/src/main/java/com/foreign/Firebase/Authentication/Email/SendVerificationEmail.java:28: error: cannot find symbol
        final FirebaseUser user = (FirebaseUser)com.foreign.Firebase.Authentication.User.GetCurrent412();
                                   ^
  symbol:   class FirebaseUser
  location: class SendVerificationEmail
/Users/caseywilliams/Fuse/QuickLaps/build/Android/Debug/app/src/main/java/com/foreign/Firebase/Authentication/Email/SendVerificationEmail.java:31: error: cannot find symbol
            .addOnCompleteListener(new OnCompleteListener<Void>() {
                                       ^
  symbol:   class OnCompleteListener
  location: class SendVerificationEmail
```

**Comment by LuisRodriguezLD** [comment]
I just made another commit, can you download and try again?

**Comment by shnupta** [comment]
```
/Users/caseywilliams/Fuse/QuickLaps/build/Android/Debug/app/src/main/java/com/foreign/Firebase/Authentication/Email/SendVerificationEmail.java:42: error: cannot find symbol
                public void onComplete(@NonNull Task<Void> task) {
                                        ^
  symbol: class NonNull
```

**Comment by shnupta** [comment]
Just need to add `"android.support.annotation.NonNull"` to the include?

**Comment by shnupta** [comment]
Big 'ol runtime error now:

```
System.err: java.lang.NoSuchMethodError: No virtual method zzckb()Z in class Lcom/google/firebase/FirebaseApp; or its super classes (declaration of 'com.google.firebase.FirebaseApp' appears in /data/app/io.github.shnupta.quicklaps-OShaXawy8SXBQf_PxZ4Svg==/base.apk)
07-20 22:53:58.637  2522  2522 W System.err: 	at com.google.firebase.database.FirebaseDatabase.getInstance(Unknown Source:117)
07-20 22:53:58.638  2522  2522 W System.err: 	at com.google.firebase.database.FirebaseDatabase.getInstance(Unknown Source:4)
07-20 22:53:58.638  2522  2522 W System.err: 	at com.foreign.Firebase.Database.DatabaseService.AndroidInit438(DatabaseService.java:42)
07-20 22:53:58.638  2522  2522 W System.err: 	at com.Bindings.ExternedBlockHost.AndroidInit438(ExternedBlockHost.java:2021)
07-20 22:53:58.638  2522  2522 W System.err: 	at com.fuse.ActivityNativeEntryPoints.cppOnCreate(Native Method)
07-20 22:53:58.638  2522  2522 W System.err: 	at com.fuse.App.onCreate(App.java:90)
07-20 22:53:58.638  2522  2522 W System.err: 	at io.github.shnupta.quicklaps.QuickLaps.onCreate(QuickLaps.java:54)
07-20 22:53:58.638  2522  2522 W System.err: 	at android.app.Activity.performCreate(Activity.java:7041)
07-20 22:53:58.638  2522  2522 W System.err: 	at android.app.Activity.performCreate(Activity.java:7032)
07-20 22:53:58.638  2522  2522 W System.err: 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1231)
07-20 22:53:58.638  2522  2522 W System.err: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2864)
07-20 22:53:58.638  2522  2522 W System.err: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2989)
07-20 22:53:58.638  2522  2522 W System.err: 	at android.app.ActivityThread.-wrap11(Unknown Source:0)
07-20 22:53:58.638  2522  2522 W System.err: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1688)
07-20 22:53:58.638  2522  2522 W System.err: 	at android.os.Handler.dispatchMessage(Handler.java:106)
07-20 22:53:58.638  2522  2522 W System.err: 	at android.os.Looper.loop(Looper.java:164)
07-20 22:53:58.638  2522  2522 W System.err: 	at android.app.ActivityThread.main(ActivityThread.java:6753)
07-20 22:53:58.638  2522  2522 W System.err: 	at java.lang.reflect.Method.invoke(Native Method)
07-20 22:53:58.638  2522  2522 W System.err: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:482)
07-20 22:53:58.639  2522  2522 W System.err: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by LuisRodriguezLD** [comment]
Fixed and merged #86
